nexus/src/db/datastore/silo.rs

```diff
@@ -188,6 +190,50 @@ impl DataStore {
 
         info!(opctx.log, "deleted {} silo users for silo {}", updated_rows, id);
 
+        // delete all silo group memberships
+        let updated_rows =
+            diesel::delete(silo_group_membership::dsl::silo_group_membership)
+                .filter(
+                    silo_group_membership::dsl::silo_group_id.eq_any(
+                        silo_group::dsl::silo_group
+                            .filter(silo_group::dsl::silo_id.eq(id))
+                            .filter(silo_group::dsl::time_deleted.is_not_null())
+                            .select(silo_group::dsl::id),
+                    ),
+                )
+                .execute_async(self.pool_authorized(opctx).await?)
+                .await
+                .map_err(|e| {
+                    public_error_from_diesel_pool(
+                        e,
+                        ErrorHandler::NotFoundByResource(authz_silo),
```

Comment:
I _think_ this should be `ErrorHandler::Server`?  (Wouldn't `NotFoundByResource` cause this to fail if the "delete" updates no rows?)

Reply:
you're right, changed in bb9f5671